syntax = "proto3";

import "file.proto";
package goya.fs;

option cc_generic_services = true;

message CreateFileRequest {
  uint64 sequence_id = 1;
  int32  type = 2;
  string file_name =3;
}

message CreateFileResponse {
  uint64 sequence_id = 1;
  int32  status = 2;
}

message ListDirectoryRequest {
  uint64 sequence_id = 1;
  string path = 2;
}

message ListDirectoryResponse {
  uint64 sequence_id = 1;
  int32  status = 2;
  repeated FileInfoProto file_info_proto = 3;
}

service MasterServer {
  rpc CreateFile(CreateFileRequest) returns(CreateFileResponse);
  rpc ListDirectory(ListDirectoryRequest) returns(ListDirectoryResponse);
}


